apps/aeoracle/src/aeo_query_tx.erl

```diff
@@ -215,20 +215,34 @@ for_client(#oracle_query_tx{sender        = SenderPubKey,
 %% -- Local functions  -------------------------------------------------------
 
 check_interaction(Q, Trees, Height) ->
-    Oracles = aec_trees:oracles(Trees),
-    Id      = aeo_interaction:id(aeo_interaction:new(Q, Height)),
-    case aeo_state_tree:lookup_interaction(Id, Oracles) of
+    Oracles  = aec_trees:oracles(Trees),
+    I        = aeo_interaction:new(Q, Height),
+    OracleId = aeo_interaction:oracle_address(I),
+    Id       = aeo_interaction:id(I),
+    case aeo_state_tree:lookup_interaction(OracleId, Id, Oracles) of
         none       -> ok;
         {value, _} -> {error, oracle_interaction_already_present}
     end.
 
-check_oracle(OraclePubKey, Trees, QueryFee) ->
+check_oracle(OraclePubKey, Trees, QueryFee, Height, QTTL, RTTL) ->
     OraclesTree  = aec_trees:oracles(Trees),
     case aeo_state_tree:lookup_oracle(OraclePubKey, OraclesTree) of
         {value, Oracle} ->
             case QueryFee >= aeo_oracles:query_fee(Oracle) of
-                true  -> ok;
+                true  -> check_oracle_ttl(Oracle, Height, QTTL, RTTL);
                 false -> {error, query_fee_too_low}
             end;
         none -> {error, oracle_does_not_exist}
     end.
+
+check_oracle_ttl(O, Height, QTTL, RTTL) ->
```

Comment:
In type aeo_oracles:fixed_ttl(), could you please replace integer() with height() (from common.hrl) or non_neg_integer()? (for readability)

In type aeo_oracles:relative_ttl(), could you please replace integer() with whether it is meant to be pos - neg - nonpos- nonneg? (for readability)